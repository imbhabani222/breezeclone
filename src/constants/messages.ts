const messages = {
  NOT_AUTHORIZED:
    "You are not authorized or session has been expired. Please try again.",
  NO_DATA_AVAILABLE: "No data available.",
  ACCESS_DENIED: "Access Denied/Forbidden.",
  DEFAULT_ERROR_MESSAGE:
    "There seems to be a problem, please retry after sometime.",
  USER_NOT_FOUND: "Invalid user",
  IMAGE_NOT_AVAILABLE: "Image not available",
  ARE_YOU_SURE: "Are you sure?",
  TYPE_OUTLINE: "outline",
  TYPE_RECTFILLED: "rectFilled",
  TYPE_FILLED: "filled",
  PLEASE_TYPE_ANSWER: "Please type your answer.",
  EXAM_NAME: "[0-2 years] Data Analytics",
  EXAM_TYPE: "By DoSelect UI/UX Testing",
  MODAL_CONTENT:
    "# Are you sure ? \n You still have **%hours% hours %minutes% mins %seconds% seconds** left to submit this assessment. </br></br>You have attempted %completed% of %total% questions.</br></br>You will not be able to attempt any question or make changes to your solution once you submit.",
  LOADING: "Loading...",
  PROCEED_TO_FEEDBACK: "PROCEED TO FEEDBACK PAGE",
  GOING_GREAT:
    "Going great .</br></br>You have attempted %completed% of %total% questions.",
  AUTO_NEXT_MODAL:
    "This section has been auto-submitted as the allotted time is up.</br></br>You have attempted %completed% of %total% questions.",
  BROWSER_TOLERANCE_INFO:
    "  You had navigated away from the test screen. This incident will be reported to the test administrator.",
  LOAD_MORE: "Load More",
  BACKBUTTON: " Go back to Assessment",
  PLATFORM_SUPPORT: "Platform Support",
  CLIENTS_SUPPORT: "Clients Support",
  HELP: "Help",
  QUESTION: "View All Questions",
  RIGHT: "right",
  FEEDBACK: {
    FEEDBACKCOMPLETED: "Woot! You've completed this assessment.",
    FEEDBACKTITLE: "SHARE FEEDBACK",
    FEEDBACKNOTIFIED:
      "We've notified team %orgName% about this. If you have any queries regarding this, please get in touch with your point of contact.",
    SENDFEEDBACK: "Send Feedback",
    ENTERFEWWORDS: "Please enter few words",
    MESSAGE_LABEL: "message",
    SUBMIT_LABEL: "Submit",
    FEEDBACKFORM: "feedbackForm",
    LIKETOTELL: "Is there something that you'd like to tell us?",
  },
  REPORT_LABLE: "REPORT AN ISSUE",
  REPORT_MODAL_TITLE: "Report an issue with Stack Using Interface",
  REPORT_MODAL_INPUT_LABEL: "  Other:",
  PROGRESS_DATABAR_TEXT: "%completed%/%total% Questions attempted",
  RUN_CODE_LABEL: "Run Code",
  ABORT_LABEL: "Abort",
  TEST_RESULTS_LABEL: "Test results",
  TEST_RESULTS_TEXT: 'Click on "Run code" to evaluate your solution',
  CUSTOM_INPUT_LABEL: "Custom Input",
  CUSTOM_INPUT_TEXT: "STDIN",
  ICON: "Icon",
  TIME_LABEL: {
    H: "H:",
    M: "M:",
    S: "S",
  },
  FAQ: {
    KEY1: "1",
    KEY2: "2",
    KEY3: "3",
    TAB1: "Help",
    TAB2: "Coding Environment",
    TAB3: "Status Code",
    SEARCH_PLACEHOLDER: "Ask a question",
    FAQINFOTEXT: `Please find below the general instructions and information about 
    DoSelect's assessment environment. We highly recommend that you go
    through this before you start taking the assessment.`,
  },
  TESTSTATUS: {
    ICONSTATUS: "iconsStatus",
    ACTIVATEICON: "activateIcon",
    DEACTIVATEICON: "deactivateIcon",
  },
  DESCRIPTION_LABEL: "Description",
  STATUS_CODE_LABEL: "Status Code",
  LIBRARIES_AVAILABLE_LABEL: "Libraries Available",
  TIME_LIMIT_LABEL: "Time Limit(in seconds)",
  MEMORY_UNIT_LABEL: "Memory Limit (in MB)",

  LIBRARIES_AVAILABLE_KEY: "Libraries available",
  TIME_LIMIT_KEY: "Time limit (in seconds)",
  MEMORY_UNIT_KEY: "Memory limit (in MB)",

  VERSION_LABEL: "Version",
  LANGUAGE_LABEL: "Language",
  DO_SELECT_RESEVE_RIGHTS: "Doselect. All rights reserved",
  LOGO_LABEL: "Logo",
  TEXT_LABEL: "text",
  TIME_FOR_EXAM_LABEL: "%hours% hours %minutes% mins",
  TIME_FOR_MODAL: "%hours% hours %minutes% mins %seconds% seconds",
  ALT_IMG: "icon Doselect",
  CANCEL: "Cancel",
  SUBMIT_END: "Submit and end the Assessment",
  SUBMIT_MOVE_NEXT: "Submit and Move to Next Section",
  PROCEED_TO_NEXT: "Proceed to next section",
  UPLOAD: "Upload",
  SAVE: "Save",
  QUESTION_TYPE: "MCQ + Coding + Fill in the  Blanks",
  LOGO: "do select logo",
  CAMERA: "Video Proctoring",
  NETWORK: "Network quality",
  TIMER: "Timer",

  BROWSER_FULLSCREEN: `<p>This %type% requires your browser window to be in fullscreen mode</p>`,
  UPLOAD_ATTACHMENT: "Upload an attachment with this solution.",
  VIDEO_TEST: {
    DESC: "Description",
    VID_DUR: "Video Duration",
    SEC: "seconds",
    RECORD: "Record",
    STOP: "Stop",
    RESUME: "Resume",
    PAUSE: "Pause",
    SUBMIT: "Submit",
    REC_DUR: "Record duration:",
  },

  CODING: {
    CODING: "Coding",
    RELOAD_STUB: "Reload Code Stub?",
    RELOAD_STUB_CONTENT: `You have changed the technology of this solution. Do you want
    to REPLACE your current code with the default code stub of
    this technology? Please note that you will lose your current
    code by doing this.`,
    NO_TEST_FOUND: "No Test Cases Found",
    RUN_ALL_TEST_CASES: "Run all Test Cases",
    DESCRIPTION: "Description",
    DESCRIPTIONS: "Descriptions",
    INPUT: "Input",
    OUTPUT: "Output",
    EXPECTED_OUTPUT: "Expected Output",
    EVAL_TIME: "Eval Time",
    CPU: "CPU",
    MEMORY: "Memory",
    EXECUTION_LOGS: "Execution Logs",
    ANNOTATION: "Annotation",
  },
};

export default messages;
