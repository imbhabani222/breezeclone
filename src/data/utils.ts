import moment from "moment";
import map from "lodash/map";
import _filter from "lodash/filter";
import { useNavigate } from "react-router-dom";
import { useState } from "react";

export const getDuration = (seconds: any, addZero?: any, callBack?: any) => {
  if (seconds < 0) return callBack ? callBack() : false;
  const duration = moment.duration(seconds, "milliseconds");
  if (addZero) {
    return {
      hours: ("0" + duration.hours()).substr(-2),
      minutes: ("0" + duration.minutes()).substr(-2),
      seconds: ("0" + duration.seconds()).substr(-2),
    };
  } else {
    return {
      hours: duration.hours(),
      minutes: duration.minutes(),
      seconds: duration.seconds(),
    };
  }
};

export const getTimeLeft = (
  timeDiff: number,
  _duration: number,
  callBack: any
) => {
  const duration = moment
    .duration(_duration, "minutes")
    .subtract(timeDiff, "seconds");
  if (duration.valueOf() <= 0) return callBack();
  return duration;
};

export const formatDateTime = (date: string | number) => {
  const _date = moment(date);
  return {
    date: _date.format("DD MMM YYYY"),
    time: _date.format("h:MM A"),
  };
};

export const getTimeDiff = (time: any) => {
  const currentTime = moment().valueOf();
  return (currentTime - time) / 1000;
};
export const stringFiller = (str: any, data: any) => {
  Object.keys(data).forEach(
    (key: any) =>
      (str = str.replaceAll(
        `%${key}%`,
        data[key]?.toString().length === 1 ? "0" + data[key] : data[key]
      ))
  );
  return str;
};

export const stringFillerInput = (str: any, data: any) => {
  console.log(str);
  Object.keys(data).forEach(
    (key: any) =>
      (str = str.replaceAll(
        `{${key}}`,
        data[key]?.toString().length === 1 ? "0" + data[key] : data[key]
      ))
  );
  return str;
};

export const checkSpace = (event: any) => {
  if (!/^\S+$/.test(event.key) && event.key !== "Backspace") {
    return true;
  }
};

export const getDurationLeft = (date: any, duration: number) => {
  const diff = getTimeDiff(date);
  const _duration = moment
    .duration(duration, "minutes")
    .subtract(diff, "seconds");
  // console.log({
  //   hours: _duration.hours(),
  //   minutes: _duration.minutes(),
  //   seconds: _duration.seconds(),
  // });

  if (duration.valueOf() <= 0) return { hours: 0, minutes: 0, seconds: 0 };
  return {
    hours: _duration.hours(),
    minutes: _duration.minutes(),
    seconds: _duration.seconds(),
  };
};

export const getAllProblemTypes = (data: any) => {
  const mapData = map(data, "problem_type");

  const names =
    mapData &&
    mapData?.filter(function (item: any, pos: any) {
      return mapData.indexOf(item) === pos;
    });

  if (names) {
    let types = names.toString().replaceAll(",", "+");
    let allTypes = types
      .replaceAll("SUB", "Subjective")
      .replaceAll("VID", "Video")
      .replaceAll("SCR", "Coding")
      .replaceAll("FIB", "Fill in the blanks");
    return allTypes;
  }
};

export const getAllSolvedIds = (data: any) => {
  let solMap = data && data?.solutions_map;
  const getKeys = Object.keys(solMap);
  return getKeys;
};

export const onSubmitFun = (
  allQuestions: any,
  mcqIndex: any,
  navi: any
  // cb: any
) => {
  let problemSlugParam = window.location.pathname.split("/")[4];
  let testSlug = window.location.pathname.split("/")[2];
  let totalProblems = allQuestions.length;

  let getIndex = allQuestions?.findIndex(
    (data: any) => data.slug === problemSlugParam
  );

  if (allQuestions.length === totalProblems) {
    navi("/test/:slug/problems".replace(":slug", testSlug));
  }

  if (
    (mcqIndex || getIndex) >= 0 &&
    (mcqIndex || getIndex) < allQuestions.length - 1
  ) {
    mcqIndex = getIndex + 1;
    navi(
      "/test/:slug/problems/:id"
        .replace(":slug", testSlug)
        .replace(":id", allQuestions[getIndex + 1].slug)
    );
  }
  //  cb();
};
