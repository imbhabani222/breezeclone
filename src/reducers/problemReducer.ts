import { combineReducers } from "redux";
import actionTypes from "../constants/actionTypes";
import {
  ProblemAction,
  PostTestAction,
  ProblemState,
  PostTestState,
  SolutionSetState,
  SolutionSetAction,
  PostSectionAction,
} from "../types/problemActionTypes";

const initialStateProblem: ProblemState = {
  pending: false,
  allProblems: "",
  error: null,
};

const postTestState: PostTestState = {
  pending: false,
  data: {},
  error: null,
};

const solutionSetState: SolutionSetState = {
  pending: false,
  allProblems: "",
  error: null,
  statusCode: null,
};
// eslint-disable-next-line import/no-anonymous-default-export
const problemReducer = (state = initialStateProblem, action: ProblemAction) => {
  switch (action.type) {
    case actionTypes.FETCH_PROBLEM_REQUEST:
      return {
        ...state,
        pending: true,
      };
    case actionTypes.FETCH_PROBLEM_SUCCESS:
      return {
        ...state,
        pending: false,
        allProblems: action.payload.allProblems,
        error: null,
      };
    case actionTypes.FETCH_PROBLEM_FAILURE:
      return {
        ...state,
        pending: false,
        allProblems: "",
        error: action.payload.error,
      };
    default:
      return {
        ...state,
      };
  }
};

const postTestReducer = (state = postTestState, action: PostTestAction) => {
  switch (action.type) {
    case actionTypes.POST_TEST_REQUEST:
      return {
        ...state,
        pending: true,
      };
    case actionTypes.POST_TEST_SUCCESS:
      return {
        ...state,
        pending: false,
        data: action.payload.data,
        error: null,
      };
    case actionTypes.POST_TEST_FAILURE:
      return {
        ...state,
        pending: false,
        error: action.payload.error,
      };
    default:
      return {
        ...state,
      };
  }
};

//
const postSectionReducer = (
  state = postTestState,
  action: PostSectionAction
) => {
  switch (action.type) {
    case actionTypes.POST_SECTION_REQUEST:
      return {
        ...state,
        pending: true,
      };
    case actionTypes.POST_SECTION_SUCCESS:
      return {
        ...state,
        pending: false,
        data: action.payload.data,
        error: null,
      };
    case actionTypes.POST_SECTION_FAILURE:
      return {
        ...state,
        pending: false,
        error: action.payload.error,
      };
    default:
      return {
        ...state,
      };
  }
};

///
const solutionSetReducer = (
  state = solutionSetState,
  action: SolutionSetAction
) => {
  switch (action.type) {
    case actionTypes.FETCH_SOLUTIONSET_REQUEST:
      return {
        ...state,
        pending: true,
        statusCode: null,
      };
    case actionTypes.FETCH_SOLUTIONSET_SUCCESS:
      return {
        ...state,
        pending: false,
        allProblems: action.payload.allProblems,
        statusCode: action.payload.statusCode,
        error: null,
      };
    case actionTypes.FETCH_SOLUTIONSET_FAILURE:
      return {
        ...state,
        pending: false,
        allProblems: "",
        error: action.payload.error,
        statusCode: action.payload.statusCode,
      };
    default:
      return {
        ...state,
      };
  }
};

export const problemsReducer = combineReducers({
  problemReducer,
  postTestReducer,
  solutionSetReducer,
  postSectionReducer,
});
