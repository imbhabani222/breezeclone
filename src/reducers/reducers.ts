import actionTypes from "../constants/actionTypes";

import { TodoActions, TodoState } from "../types/types";

const initialState: TodoState = {
  pending: false,
  todos: [],
  error: null
};

// eslint-disable-next-line import/no-anonymous-default-export
export default (state = initialState, action: TodoActions) => {
  switch (action.type) {
    case actionTypes.FETCH_TODO_REQUEST:
      return {
        ...state,
        pending: true
      };
    case actionTypes.FETCH_TODO_SUCCESS:
      return {
        ...state,
        pending: false,
        todos: action.payload.todos,
        error: null
      };
    case actionTypes.FETCH_TODO_FAILURE:
      return {
        ...state,
        pending: false,
        todos: [],
        error: action.payload.error
      };
    default:
      return {
        ...state
      };
  }
};
