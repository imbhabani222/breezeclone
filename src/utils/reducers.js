/* eslint-disable no-prototype-builtins */
export const createReducer = (initialState, handlers) => {
  return function reducer(state = initialState, action) {
    if (handlers.hasOwnProperty(action.type)) {
      return handlers[action.type](state, action);
    } else {
      return state;
    }
  };
};

export const getFetchingState = (state) => ({
  ...state,
  fetching: true,
  result: null,
  error: null,
});

export const getSuccessState = (state, { result }) => ({
  ...state,
  fetching: false,
  result: result.result || result,
  error: null,
});

export const getErrorState = (state, { error }) => ({
  ...state,
  fetching: false,
  result: null,
  error,
});

export const getResetState = (state) => ({
  ...state,
  fetching: false,
  result: null,
  error: null,
});

export const getActionInitialState = (state) => ({
  ...state,
});

export const getActionDefaultState = (state, { result: { data } }) => ({
  ...state,
  ...data,
});

export const checkAction = (state = {}, action) => {
  switch (action.type) {
    case "CHECK_ACTION": {
      console.log(state, action);
      return {
        data: { ...state.data, ...action.data },
      };
    }
    default:
      return state;
  }
};

export const customResult = (state = {}, action) => {
  switch (action.type) {
    case "GET_CUSTOM_RESULT": {
      console.log(state, action, "custommmmmmmmmm");
      return {
        data: { ...state.data, ...action.data },
      };
    }
    default:
      return state;
  }
};

export const testCasesResult = (state = {}, action) => {
  switch (action.type) {
    case "GET_TESTCASES_RESULT": {
      return {
        data: { ...state.data, ...action.data },
      };
    }
    default:
      return state;
  }
};
