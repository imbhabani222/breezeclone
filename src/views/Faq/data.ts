export const data: any = [
  [
    {
      Language: "Assembler 32 (Mars 4.5)",
      Version: "Mars 4.5",
      "Memory limit (in MB)": 512,
      "Time limit (in seconds)": 10,
      "Libraries available": null,
    },
    {
      Language: "BASH",
      Version: "4.3.11",
      "Memory limit (in MB)": 512,
      "Time limit (in seconds)": 4,
      "Libraries available": null,
    },
    {
      Language: "C",
      Version: "gcc 4.8.2, C99 standard",
      "Memory limit (in MB)": 512,
      "Time limit (in seconds)": 2,
      "Libraries available": ["JSON", "Math Lib"],
    },
    {
      Language: "C++",
      Version: "g++ 4.8.4, C++11 standard",
      "Memory limit (in MB)": 512,
      "Time limit (in seconds)": 2,
      "Libraries available": ["JSON", "Math Lib"],
    },
    {
      Language: "C#",
      Version: "Mono Compiler 4.2.1",
      "Memory limit (in MB)": 512,
      "Time limit (in seconds)": 5,
      "Libraries available": null,
    },
    {
      Language: "Clojure",
      Version: "1.6.0",
      "Memory limit (in MB)": 512,
      "Time limit (in seconds)": 8,
      "Libraries available": null,
    },
    {
      Language: "CommonLisp",
      Version: "GNU CLISP 2.49",
      "Memory limit (in MB)": 512,
      "Time limit (in seconds)": 12,
      "Libraries available": null,
    },
    {
      Language: "Erlang",
      Version: "Erlang/OTP 24 [erts-12.0.2]",
      "Memory limit (in MB)": 512,
      "Time limit (in seconds)": 12,
      "Libraries available":
        "Your module should be named as 'solution' and it should export a method named 'start'",
    },
    {
      Language: "Elixir",
      Version: "Elixir 1.12.0 (compiled with Erlang/OTP 24)",
      "Memory limit (in MB)": 512,
      "Time limit (in seconds)": 12,
      "Libraries available":
        "You must invoke your method at the end of the script",
    },
    {
      Language: "F#",
      Version: "F# Compiler for F# 4.0, Mono 4.2.1",
      "Memory limit (in MB)": 512,
      "Time limit (in seconds)": 5,
      "Libraries available": null,
    },
    {
      Language: "Go",
      Version: "1.5.3",
      "Memory limit (in MB)": 512,
      "Time limit (in seconds)": 4,
      "Libraries available": null,
    },
    {
      Language: "Haskell",
      Version: "7.6.3",
      "Memory limit (in MB)": 512,
      "Time limit (in seconds)": 5,
      "Libraries available": null,
    },
    {
      Language: "Java 7",
      Version: "1.7.0_72",
      "Memory limit (in MB)": 512,
      "Time limit (in seconds)": 10,
      "Libraries available":
        "json-simple\n <strong>Note</strong>: Use 'Source' as class name for java program",
    },
    {
      Language: "Java 8",
      Version: "1.8.0_25",
      "Memory limit (in MB)": 512,
      "Time limit (in seconds)": 10,
      "Libraries available":
        "json-simple\n <strong>Note</strong>: Use 'Source' as class name for java program",
    },
    {
      Language: "JavaScript",
      Version: "Node.js 8.1.4",
      "Memory limit (in MB)": 512,
      "Time limit (in seconds)": 10,
      "Libraries available": null,
    },
    {
      Language: "Julia",
      Version: "0.3.7",
      "Memory limit (in MB)": 512,
      "Time limit (in seconds)": 4,
      "Libraries available": null,
    },
    {
      Language: "Lua",
      Version: "5.2.3",
      "Memory limit (in MB)": 512,
      "Time limit (in seconds)": 12,
      "Libraries available": null,
    },
    {
      Language: "Objective-C",
      Version: "Clang 3.4",
      "Memory limit (in MB)": 512,
      "Time limit (in seconds)": 2,
      "Libraries available": null,
    },
    {
      Language: "PHP",
      Version: "5.5.9-1",
      "Memory limit (in MB)": 512,
      "Time limit (in seconds)": 9,
      "Libraries available": null,
    },
    {
      Language: "Perl",
      Version: "v5.18.2",
      "Memory limit (in MB)": 512,
      "Time limit (in seconds)": 9,
      "Libraries available": null,
    },
    {
      Language: "Python 2",
      Version: "2.7.13",
      "Memory limit (in MB)": 512,
      "Time limit (in seconds)": 10,
      "Libraries available": [
        "requests 2.12.0",
        "simplejson 3.10.0",
        "numpy 1.11.2",
        "scipy 0.18.0",
        "scikit-learn 0.18",
        "matplotlib 1.5.3",
        "cython 0.25.2",
        "pandas 0.19.0",
        "nltk 3.2",
        "pydataset 0.2",
        "imbalanced-learn",
        "seaborn 0.8",
        "statsmodels 0.8",
      ],
    },
    {
      Language: "Python 3",
      Version: "3.4.0",
      "Memory limit (in MB)": 512,
      "Time limit (in seconds)": 10,
      "Libraries available": [
        "requests  2.12.0",
        "simplejson 3.10.0",
        "numpy 1.11.2",
        "scipy 0.18.0",
        "scikit-learn 0.18",
        "matplotlib 1.5.3",
        "cython 0.25.2",
        "pandas 0.19.0",
        "nltk 3.2",
        "pydataset 0.2",
        "imbalanced-learn",
        "seaborn 0.8",
        "statsmodels 0.8",
      ],
    },
    {
      Language: "R",
      Version: "3.0.2",
      "Memory limit (in MB)": 512,
      "Time limit (in seconds)": 10,
      "Libraries available": [
        "RCurl",
        "dplyr",
        "tidyr",
        "readr",
        "stringr",
        "plyr",
        "reshape2",
        "lubridate",
        "data.table",
        "dtplyr",
        "foreach",
        "ggplot2",
        "RColorBrewer",
        "randomForest",
        "rpart",
        "ROCR",
        "car",
        "party",
        "caret",
        "mlr",
        "tree",
        "klaR",
        "ipred",
        "lars",
        "penalized",
        "mlbench",
        "e1071",
        "sqldf",
        "glmnet",
        "gbm",
        "knitr",
        "ggthemes",
        "ggmap",
        "gridExtra",
        "xgboost",
        "tm",
        "wordcloud2",
        "dummies",
        "mice",
        "forecast",
        "naivebayes",
        "caTools",
        "h2o",
        "ggpubr",
        "arules",
        "animation",
        "png",
        "arulesViz",
      ],
    },
    {
      Language: "Ruby",
      Version: "1.9.3p484",
      "Memory limit (in MB)": 512,
      "Time limit (in seconds)": 10,
      "Libraries available": null,
    },
    {
      Language: "Rust",
      Version: "1.5.0",
      "Memory limit (in MB)": 512,
      "Time limit (in seconds)": 5,
      "Libraries available": null,
    },
    {
      Language: "Scala",
      Version: "2.11.4",
      "Memory limit (in MB)": 512,
      "Time limit (in seconds)": 7,
      "Libraries available": null,
    },
    {
      Language: "Swift",
      Version: "4.0.2",
      "Memory limit (in MB)": 512,
      "Time limit (in seconds)": 5,
      "Libraries available": null,
    },
  ],
];
