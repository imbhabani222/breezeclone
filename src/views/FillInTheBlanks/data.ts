import exclamationCircle from "../../assets/exclamation_circle.svg";
import clearSvg from "../../assets/clear.svg";
import listIcon from "../../assets/list_item.svg";
export const dataFillInBlanksFun = ({ onOk, onCancel, disabled }: any) => {
  return {
    typeOfQuestions: "Fill In the Blanks",
    Icon: listIcon,
    // title: "Answer",
    //text: "Please type your answer.",
    buttons: [
      {
        type: "rectFilled",
        display: "REPORT AN ISSUE",
        onClick: onOk,
        beforeIcon: exclamationCircle,
      },
      {
        type: "rectOutline",
        display: "CLEAR",
        disabled: disabled,
        onClick: onCancel,
        beforeIcon: clearSvg,
      },
    ],
  };
};
export const FooterButtons = ({
  onClick,
  disabledsave,
  disabledsubmit,
  onSubmit,
}: any) => {
  return [
    {
      type: "rectFilled",
      display: "Save",
      onClick: onClick,
      disabled: disabledsave,
    },
    {
      type: "rectFilled",
      display: "Submit",
      onClick: onSubmit,
      disabled: disabledsubmit,
    },
  ];
};
export const dataVideoTest = ({ onOk, onCancel, disabled }: any) => {
  return {
    typeOfQuestions: "Record Video",
    hideIcon: true,
    // title: "Answer",
    buttons: [
      {
        type: "rectFilled",
        display: "REPORT AN ISSUE",
        onClick: onOk,
        beforeIcon: exclamationCircle,
      },
    ],
  };
};
