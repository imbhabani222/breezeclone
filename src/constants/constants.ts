import Cookies from "universal-cookie";
import messages from "./messages";

import { useLocation } from "react-router-dom";
const cookies = new Cookies();
const csrfToken = cookies.get("doselectcsrf");

const constants = {
  HTTP_REQ_DEFAULT_HEADERS: {
    Accept: "application/json",
    "Content-Type": "application/json; charset=UTF-8",
  },
  HTTP_POST_HEADERS: {},
  HTTP_MULTIPART_FORM_HEADERS: {
    "Content-Type": false,
  },
  HTTP_METHODS: {
    POST: "POST",
    GET: "GET",
    DELETE: "DELETE",
    PATCH: "PATCH",
    PUT: "PUT",
  },
  HTTP_REQ_ERROR_MAP: {
    401: { status: 401, message: messages.NOT_AUTHORIZED },
    403: { status: 403, message: messages.ACCESS_DENIED },
    404: { status: 404, message: messages.NO_DATA_AVAILABLE },
  },
  ROUTE: {
    LANDING_PAGE: "/test/:slug/landingPage",
    INSTRUCTION_PAGE: "/test/:slug/instructionPage",
    DOSELECT_INSTRUCTION: "/test/:slug/doSelectInstruction",
    TEST_SECTION_SUBMIT: "/test/:slug/problems",
    FAQ: "/test/:slug/Faqs",
    FEEDBACK: "/test/:slug/Feedback",
    TESTCOMPLETE: "/test/:slug/TestComplete",
    TESTLAYOUT: "/test/:slug/TestLayout",
    RICH_TEXT_EDITOR: "/test/:slug/problems/id",
    FIll_IN_BLANKS: "/tegst/:slug/problems/:id",
    MCQ_PAGE: "/tegst/:slug/problems/:id",
    SWITCH_PATH: "/test/:slug/problems/:id",
  },
  NAME_KEY: "name",
  STATUS_KEY: "status",
  LANGUAGE_KEY: "Language",
  VERSION_KEY: "Version",
  MU_KEY: "Mu",
  TL_KEY: "Tl",
  LA_KEY: "LA",
  DESCRIPTION_KEY: "description",
  DEFAULT_KEY: "default",
  VIDEO_KEY: "video",
  PROGRESSBAR_KEY: "progressbar",
};

export default constants;
