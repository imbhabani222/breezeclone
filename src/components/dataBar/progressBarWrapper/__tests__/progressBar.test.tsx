import React from "react";
import { BrowserRouter } from "react-router-dom";
import { cleanup, fireEvent, render, screen } from "@testing-library/react";
import ProgressDataBarWrapper from "../progressBarWrapper";
import constants from "../../../../constants/constants";
import { combineReducers, createStore } from "redux";
import messages from "../../../../constants/messages";
import { problemsReducer } from "../../../../reducers/problemReducer";
import { stringFiller } from "../../../../data/utils";
import { Provider } from "react-redux";

const { PROGRESSBAR_KEY } = constants;
const { PROGRESS_DATABAR_TEXT } = messages;
describe("ProgressBar Testing", () => {
  const mock = [
    {
      name: "Section 1",
      status: "",
      problems: [
        {
          id: 320603,
          name: "Dot Product M",
          problem_type: "MCQ",
          slug: "83p8o",
          solution: [1],
        },
      ],
    },
    {
      name: "Section 2",
      status: "",
      problems: [
        {
          id: 320603,
          name: "Dot Product M",
          problem_type: "MCQ",
          slug: "83p8o",
          solution: null,
        },
      ],
    },
  ];
  const initialState = {
    pending: false,
    allProblems: mock,
    error: null,
  };

  const fakeStore = createStore(
    combineReducers({ problem: problemsReducer } as any),
    {
      problem: initialState,
    }
  );
  console.log(fakeStore, "fakestore");
  fakeStore.dispatch = jest.fn((): any => {
    return mock;
  });

  function renderProgressDatabar() {
    return render(
      <Provider store={fakeStore}>
        <BrowserRouter>
          <ProgressDataBarWrapper />
        </BrowserRouter>
      </Provider>
    );
  }

  afterEach(cleanup);
  test("Render without Crashing", () => {
    renderProgressDatabar();
  });
  test("snapshot button component", () => {
    const { asFragment } = renderProgressDatabar();
    expect(asFragment()).toMatchSnapshot();
  });
  test("should display progressdatabar", async () => {
    renderProgressDatabar();
    // expect(screen.queryByText(PROGRESSBAR_KEY)).not.toBeInTheDocument();
    const wrap = stringFiller(PROGRESS_DATABAR_TEXT, {
      completed: 1,
      total: 1,
    });
    expect(screen.getByTitle("databartitle").innerHTML).toMatch(wrap);
  });

  //   test("display databar", async () => {
  //     renderProgressDatabar();
  //     expect(screen.getByText()).toBeInTheDocument();
  //   });
});
