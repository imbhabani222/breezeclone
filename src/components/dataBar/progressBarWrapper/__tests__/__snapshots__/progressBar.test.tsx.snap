// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar Testing snapshot button component 1`] = `
<DocumentFragment>
  <div
    class="progressbar_col"
  >
    <div
      class="ant-progress ant-progress-line ant-progress-status-normal ant-progress-default progress"
    >
      <div
        class="ant-progress-outer"
      >
        <div
          class="ant-progress-inner"
        >
          <div
            class="ant-progress-bg"
            style="width: 0%; height: 10px;"
          />
        </div>
      </div>
    </div>
    <span
      class="ant-typography progressbar_text"
      title="%completed%/%total% Questions attempted"
    >
      00/00 Questions attempted
    </span>
  </div>
</DocumentFragment>
`;
